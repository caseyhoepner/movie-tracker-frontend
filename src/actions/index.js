export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

export const toggleFavorite = (movieId) => ({
  type: 'TOGGLE_FAVORITE',
  movieId: movieId
})

export const setUser = (userId) => ({
  type: 'SET_USER',
  userId: userId
})

export const removeUser = () => ({
  type: 'REMOVE_USER'
})

export const filterMovies = (movies) => ({
  type: 'FILTER_MOVIES',
  movies
})