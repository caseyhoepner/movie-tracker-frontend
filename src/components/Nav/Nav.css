.nav-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  width: 200px;
  font-size: 1.25rem;
}

.fav-btn-group {
  /*text should be display: none until :hover, then CSS transition ease in above the respective icon;*/
}

.fav-btn-nav {
  height: 50px;
  width: 50px;
  background-image: url('../../assets/images/hatFull-red.svg');
  background-size: cover;
}

.num-favs {
  position: relative;
  top: 5px;
  left: 25px;
}

.favorites-text,
.login-text {
  color: #D4A931;
  visibility: hidden;
}

.active p {
  position: relative;
  bottom: 45px;
  right: 50px;
  visibility: visible;
  transition: visibility 2s ease-in-out;
}

.active.login-text {
  bottom: 45px;
  right: 50px;
}

/*.active.favorites-text {
  bottom: 60px;
  right: 40px;
}*/

.login-btn {
  width: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.login-icon {
  height: 50px;
  background-image: url('../../assets/images/margot.png');
  background-size: cover;
}

a {
  color: #D4A931;
  text-decoration: none;
}