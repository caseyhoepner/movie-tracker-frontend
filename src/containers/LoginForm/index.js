import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  render() {
    return(
      <form>
        <input />
      </form>
    )
  }
}

export default LoginForm;