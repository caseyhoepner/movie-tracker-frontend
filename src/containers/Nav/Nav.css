.nav-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  width: 200px;
  font-size: 1.25rem;
}

.fav-btn-group {
  /*text should be display: none until :hover, then CSS transition ease in above the respective icon;*/
}

.fav-btn-nav {
  height: 50px;
  width: 50px;
  background-image: url('../../assets/images/hatFull-red.svg');
  background-size: cover;
}

.num-favs {
  position: relative;
  top: 5px;
  left: 25px;
}

.favorites-text,
.login-text {
  position: relative;
  color: #D4A931;
  opacity: 0;
}

a:hover p {
  opacity: 1;
}

a:hover .login-text {
  bottom: 50px;
  right: 50px;
}

a:hover .favorites-text {
  bottom: 50px;
  right: 25px;
}

.login-text {
  bottom: 30px;
  right: 50px;
  transition: 1s;
}

.favorites-text {
  bottom: 30px;
  right: 25px;
  transition: 1s;
}

.login-btn {
  width: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.login-icon {
  height: 50px;
  background-image: url('../../assets/images/margot.png');
  background-size: cover;
}

a {
  color: #D4A931;
  text-decoration: none;
}