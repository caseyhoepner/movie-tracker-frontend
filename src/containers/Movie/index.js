import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';

import * as API from '../../utils'
import { toggleFavorite } from '../../actions';
import './Movie.css';

// make functional
export class Movie extends Component {
  handleClick = async () => {
    const { movie_id, overview, poster_path, release_date, title, vote_average, isLoggedIn, toggleFavorite } = this.props;
    const movie = {
      user_id: isLoggedIn,
      movie_id: movie_id,
      overview,
      poster_path,
      release_date,
      title,
      vote_average,
    }

    await API.addFavorite(movie)
    toggleFavorite(movie_id)
  }

  render() {
    let favoriteBtn;
    const { poster_path, movie_id, isLoggedIn, isFavorite } = this.props;

    if (isLoggedIn) {
      favoriteBtn =
        <button className='m-fav-container'
          onClick={this.handleClick}
        >
          <p className='m-fav-text'>FAVORITE</p>
          <div
            className={`m-fav-btn ${isFavorite ? 'red' : ''}`}>
          </div>
        </button>
    } else {
      favoriteBtn =
        <NavLink className='m-fav-container'
          to='/login'>
          <p className='m-fav-text'>FAVORITE</p>
          <div className='m-fav-btn'>
          </div>
        </NavLink>
    }

    return (
      <div className='m-movie'>
        <Link to={`/movie/${movie_id}`}>
          <div className='m-movie-poster'>
            <img
              className='m-poster'
              src={`http://image.tmdb.org/t/p/w342//${poster_path}`}
              alt='Click here for movie details.'/>
          </div>
        </Link>
        { favoriteBtn }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  isLoggedIn: state.user.id,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId)),
})

Movie.propTypes = {
  movie_id: PropTypes.number.isRequired,
  overview: PropTypes.string.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote_average: PropTypes.number,
  isFavorite: PropTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);