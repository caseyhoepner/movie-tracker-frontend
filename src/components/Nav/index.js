import React from 'react';
import PropTypes from 'prop-types';
import './Nav.css';
import { NavLink } from 'react-router-dom';
import userIcon from '../../assets/images/margot.png'

const Nav = () => {
  return (
    <nav className="nav-container">
      <NavLink
        className="fav-btn-group"
        to="/favorites">
        <div className="fav-btn-nav">
          <span className="num-favs">0</span>
        <p className="favorites-text">FAVORITES</p>
        </div>
      </NavLink>
      <NavLink to="/login" className="login-btn">
        <img
          className="login-icon"
          src={ userIcon } />
        <p className="login-text">LOGIN</p>
      </NavLink>
    </nav>
  )
}

export default Nav;