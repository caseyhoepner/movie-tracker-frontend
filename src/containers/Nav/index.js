import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeUser, filterMovies } from '../../actions';
import userIcon from '../../assets/images/margot.png';
import './Nav.css';

export const Nav = (props) => {
  const { userLoggedIn, removeUser, movies, filterMovies } = props

  const handleHover = (event) => {
    if(!event.target.classList.contains('active')) {
      event.target.classList.add('active')
    } else {
      event.target.classList.remove('active')
    }
  }

  return (
    <nav className="nav-container">
      <NavLink
        onClick={() => filterMovies(movies)}
        to="/favorites"
        className="fav-btn-group"
      >
        <div
          className="fav-btn-nav"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <span className="num-favs">0</span>
          <p
            className={`favorites-text`}
          >
            FAVORITES
          </p>
        </div>
      </NavLink>
      <NavLink
        to="/login"
        className={userLoggedIn ? "hidden" : "show"}
      >
        <div
          className="login-btn"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <img
            className="login-icon"
            src={ userIcon }
            alt='Click here to login'/>
          <p className="login-text">
            LOGIN
          </p>
        </div>
      </NavLink>
      <NavLink
        className={userLoggedIn ? "logout-btn" : "hidden"}
        onClick={removeUser}
        to='/'
      >
        <img
          className="login-icon"
          src={ userIcon }
          alt='Click here to login'/>
        <p className="login-text">
          LOGOUT
        </p>
      </NavLink>
    </nav>
  )
}

export const mapStateToProps = (state) => ({
  userLoggedIn: state.user.id,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  filterMovies: (movies) => dispatch(filterMovies(movies))
})

Nav.propTypes = {
  removeUser: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.number
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);