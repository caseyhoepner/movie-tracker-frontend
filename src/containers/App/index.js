import React, { Component } from 'react';
import { Link, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as API from '../../utils';
import apiKey from '../../API-key.js';
import { addMovies, toggleFavorite } from '../../actions';
import MovieContainer from '../MovieContainer';
import Nav from '../Nav';
import LoginForm from '../LoginForm'
import MovieDetails from '../../components/MovieDetails';
import wesIcon from '../../assets/images/wes.png';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="a-App">
        <header className="a-header-container">
          <Link to='/' className="a-logo">
            <img
              src={ wesIcon }
              className="a-logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1 className='a-header-title'>WesTracker</h1>
          </Link>
          <Nav />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
        <Route
          exact path='/login'
          component={ LoginForm }
        />
        <Route path='/movie/:id' render={({match}) => {
          const { id } = match.params;
          const movie = this.props.movies.find(movie => (
            movie.id === parseInt(id, 10)))
          return (
            <MovieDetails {...movie} />
          )}} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId))
})

App.propTypes = {
  user: PropTypes.object.isRequired,
  addMovies: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));