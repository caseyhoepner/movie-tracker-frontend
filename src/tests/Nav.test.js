import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapDispatchToProps, mapStateToProps } from '../containers/Nav';
import * as Actions from '../actions'

describe('Nav', () => {
  describe('Nav Component', () => {
    let wrapper;
    let removeUser;

    beforeEach(() => {
      removeUser = jest.fn()
      wrapper = shallow(<Nav
                          userLoggedIn={undefined}
                          removeUser={removeUser}
                        />);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should toggle login text depending on user login status', () => {
      wrapper = shallow(<Nav
                          userLoggedIn={1}
                          removeUser={removeUser}
                        />)

      expect(wrapper).toMatchSnapshot()
    })

  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch with the removeUser action', () => {
      const mockDispatch = jest.fn()
      const expected = Actions.removeUser()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removeUser()

      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the userLoggedIn from state', () => {
      const mockState = {
        user: {
          id: 1
        }
      }

      const expected = 1

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.userLoggedIn).toEqual(expected)
    })
  })
})