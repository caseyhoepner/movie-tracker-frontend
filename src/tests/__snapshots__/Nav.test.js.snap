// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav Component should match the snapshot 1`] = `
<nav
  className="nav-container"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="fav-btn-group"
    onClick={[Function]}
    to="/favorites"
  >
    <div
      className="fav-btn-nav"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <span
        className="num-favs"
      >
        0
      </span>
      <p
        className="favorites-text"
      >
        FAVORITES
      </p>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="show"
    to="/login"
  >
    <div
      className="login-btn"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Click here to login"
        className="login-icon"
        src="margot.png"
      />
      <p
        className="login-text"
      >
        LOGIN
      </p>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="hidden"
    onClick={[MockFunction]}
    to="/"
  >
    <img
      alt="Click here to login"
      className="login-icon"
      src="margot.png"
    />
    <p
      className="login-text"
    >
      LOGOUT
    </p>
  </NavLink>
</nav>
`;

exports[`Nav Nav Component should toggle login text depending on user login status 1`] = `
<nav
  className="nav-container"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="fav-btn-group"
    onClick={[Function]}
    to="/favorites"
  >
    <div
      className="fav-btn-nav"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <span
        className="num-favs"
      >
        0
      </span>
      <p
        className="favorites-text"
      >
        FAVORITES
      </p>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="hidden"
    to="/login"
  >
    <div
      className="login-btn"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Click here to login"
        className="login-icon"
        src="margot.png"
      />
      <p
        className="login-text"
      >
        LOGIN
      </p>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="logout-btn"
    onClick={[MockFunction]}
    to="/"
  >
    <img
      alt="Click here to login"
      className="login-icon"
      src="margot.png"
    />
    <p
      className="login-text"
    >
      LOGOUT
    </p>
  </NavLink>
</nav>
`;
